velox/exec/HashTable.h

```diff
@@ -95,6 +95,11 @@ struct HashLookup {
   /// If using valueIds, list of concatenated valueIds. 1:1 with 'hashes'.
   /// Populated by groupProbe and joinProbe.
   raw_vector<uint64_t> normalizedKeys;
+
+  /// If true, the probe will only consider rows with null keys.
+  /// During probing, only the hash value is compared, and key comparison is
+  /// skipped to avoid potential issues. This is used by listNullKeyRows.
```

Comment:
> key comparison is skipped to avoid potential issues

Just curious, why would `hashers_[0]->decodedVector().base() ` be a dangling pointer ? is there some issue with the lifetime of the vector passed to this hasher? OR do we never pass any vector to it for decoding?

Just want to make sure we dont end up masking any inconsistent state in our data structures

Reply:
`hashers_` comes from the hash table, and it should be used during the hash build phase. Also, `baseVector_` (`hashers_[0]->decodedVector().base()`) is a raw pointer, so we cannot guarantee that `baseVector_` is accessible during probe phase. I think in `listNullKeyRows`, comparing only the hash value should be sufficient to find null keys.

Reply:
> baseVector_ (hashers_[0]->decodedVector().base()) is a raw pointer, so we cannot guarantee that baseVector_ is available during probe phase.

A decodedVector is intended for reading existing vectors, and we should ensure its validity if it's accessible. The fact that it was a dangling pointer suggests some action released those vectors while still allowing access, which this PR aims to fix. However, it would be beneficial to investigate the root cause of this inconsistent state to prevent future potential bugs and ensure we're not just addressing this specific symptom.

Reply:
As I mentioned above, the vector whose pointer becomes dangling here should be an input vector for HashBuild. It seems a bit odd to me that during the hash probe phase, we need to access the build-side input vector. From my understanding, the vector being released is expected behavior by design. The issue here is that we should not be attempting to access it.